src/renderers/dom/shared/ReactDOMTextComponent.js

```diff
@@ -48,6 +48,7 @@ var ReactDOMTextComponent = function(text) {
   this._mountIndex = 0;
   this._closingComment = null;
   this._commentNodes = null;
+  this._isWhitespace = typeof text === 'string' && text.trim().length === 0;
```

Comment:
`text` in this case is not ensured to be a string. I'm also not sure if adding a property here is the right move. Maybe the check should be done in `validateDOMNesting.js`?